src/Components/Server/test/Circuits/CircuitMetricsTest.cs

```diff
@@ -0,0 +1,204 @@
+        using var circuitDurationCollector = new MetricCollector<double>(_meterFactory,
+            CircuitMetrics.MeterName, "aspnetcore.components.circuits.duration");
+
+        // Act
+        var startTime = Stopwatch.GetTimestamp();
+        Thread.Sleep(10); // Add a small delay to ensure a measurable duration
+        var endTime = Stopwatch.GetTimestamp();
+        circuitMetrics.OnCircuitDown(startTime, endTime);
+
+        // Assert
+        var activeMeasurements = activeCircuitCounter.GetMeasurementSnapshot();
+        var connectedMeasurements = connectedCircuitCounter.GetMeasurementSnapshot();
+        var durationMeasurements = circuitDurationCollector.GetMeasurementSnapshot();
+
+        Assert.Single(activeMeasurements);
+        Assert.Equal(-1, activeMeasurements[0].Value);
+
+        Assert.Single(connectedMeasurements);
+        Assert.Equal(-1, connectedMeasurements[0].Value);
+
+        Assert.Single(durationMeasurements);
+        Assert.True(durationMeasurements[0].Value > 0);
+    }
+
+    [Fact]
+    public void IsDurationEnabled_ReturnsMeterEnabledState()
+    {
+        // Arrange
+        var circuitMetrics = new CircuitMetrics(_meterFactory);
+
+        // Create a collector to ensure the meter is enabled
+        using var circuitDurationCollector = new MetricCollector<double>(_meterFactory,
+            CircuitMetrics.MeterName, "aspnetcore.components.circuits.duration");
+
+        // Act & Assert
+        Assert.True(circuitMetrics.IsDurationEnabled());
+    }
+
+    [Fact]
+    public void FullCircuitLifecycle_RecordsAllMetricsCorrectly()
+    {
+        // Arrange
+        var circuitMetrics = new CircuitMetrics(_meterFactory);
+        using var totalCounter = new MetricCollector<long>(_meterFactory,
+            CircuitMetrics.MeterName, "aspnetcore.components.circuits.count");
+        using var activeCircuitCounter = new MetricCollector<long>(_meterFactory,
+            CircuitMetrics.MeterName, "aspnetcore.components.circuits.active_circuits");
+        using var connectedCircuitCounter = new MetricCollector<long>(_meterFactory,
+            CircuitMetrics.MeterName, "aspnetcore.components.circuits.connected_circuits");
+        using var circuitDurationCollector = new MetricCollector<double>(_meterFactory,
+            CircuitMetrics.MeterName, "aspnetcore.components.circuits.duration");
+
+        // Act - Simulating a full circuit lifecycle
+        var startTime = Stopwatch.GetTimestamp();
+
+        // 1. Circuit opens
+        circuitMetrics.OnCircuitOpened();
+
+        // 2. Connection established
+        circuitMetrics.OnConnectionUp();
+
+        // 3. Connection drops
+        circuitMetrics.OnConnectionDown();
+
+        // 4. Connection re-established
+        circuitMetrics.OnConnectionUp();
+
+        // 5. Circuit closes
+        Thread.Sleep(10); // Add a small delay to ensure a measurable duration
```

Comment:
nit: Is better to make the test async and use `Task.Delay` as that won't block the thread, doing this repeatedly will slow-down the test suite.